Enable some RSA key import export related tests.

nmahendru opened the pull request:
#### 0f246af6c3a3807e9233d89ff1f47f7d45563f30
<pre>
Enable some RSA key import export related tests.
<a href="https://rdar.apple.com/115676486">rdar://115676486</a>

Reviewed by Alex Christensen.

This updates the d value for RSA keys in various tests.
The way d was calculated changed around end of year 2019 which caused
the exported value to not match the input value.

* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker.html:
* LayoutTests/crypto/subtle/rsa-import-jwk-key-export-jwk-key-private-expected.txt:
* LayoutTests/crypto/subtle/rsa-import-jwk-key-export-jwk-key-private.html:
* Source/WebCore/crypLayoutTests/platform/ios/TestExpectations
* LayoutTests/platform/mac/TestExpectations
* Source/WebCore/crypto/mac/CryptoKeyRSAMac.cpp
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBStructuredCloneBackwardCompatibility.mm
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBStructuredCloneBackwardCompatibilityRead.html
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBStructuredCloneBackwardCompatibilityWrite.html

Canonical link: <a href="https://commits.webkit.org/272691@main">https://commits.webkit.org/272691@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/750ca9676bdbd513298af93f59d31279fe2bbf93

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32631 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11384 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34485 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35203 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29497 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33541 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13731 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8571 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28994 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33070 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9594 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29151 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8367 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8499 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29098 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36539 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29655 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29513 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34622 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8623 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6578 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32485 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10304 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/28813 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9237 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4217 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9225 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nmahendru commented on LayoutTests/crypto/subtle/aes-cbc-import-key-wrap-jwk-rsa-key-private.html:
```diff
@@ -12,6 +12,21 @@
 description("Test wrapping a JWK RSA private key with AES-CBC using an imported key");
 
 jsTestIsAsync = true;
+var jwkKeyPhiFormat = {
```
This is the exact same value as jwkKeyBefore

achristensen07 commented on Source/WebCore/crypto/mac/CryptoKeyRSAMac.cpp:
```diff
@@ -131,9 +131,14 @@ RefPtr<CryptoKeyRSA> CryptoKeyRSA::create(CryptoAlgorithmIdentifier identifier,
 
     CCRSACryptorRef cryptor = nullptr;
-    // FIXME: It is so weired that we recaculate the private exponent from first prime factor and second prime factor,
+    // FIXME(nitin: Does not need to be now): It is so weired that we recaculate the private exponent from first prime factor and second prime factor,
```
We typically don't put names in fixme comments.  Anyone can fix this.

nmahendru replied:
I have removed the fixme and updated the comment now.

achristensen07 commented on Source/WebCore/crypto/mac/CryptoKeyRSAMac.cpp:
```diff
@@ -131,9 +131,14 @@ RefPtr<CryptoKeyRSA> CryptoKeyRSA::create(CryptoAlgorithmIdentifier identifier,
     // given the fact that we have already had it. Also, the re-caculated private exponent may not match the given one.
     // See <rdar://problem/15452324>.
+    //    I spent some time to fix this and the CCRSACryptorRecoverPrivateKey in CommonCrypto actually works with n,e,d but
+    //    I also saw that chrome and firefox don't support just d without additional params.
```
We also typically don't use first person, it being unclear who "I" is referring to.  

CCRSACryptorRecoverPrivateKey in CommonCrypto actually works with n,e,d but
Chrome and Firefox don't support just d without additional params.

nmahendru replied:
comment updated.

achristensen07 commented on Source/WebCore/crypto/openssl/CryptoKeyRSAOpenSSL.cpp:
```diff
@@ -45,17 +45,9 @@ static size_t getRSAModulusLength(RSA* rsa)
-    CryptoKeyType keyType;
-    switch (keyData.type()) {
-    case CryptoKeyRSAComponents::Type::Public:
+    CryptoKeyType keyType = CryptoKeyType::Private;
```
This change doesn't seem meaningful.  If anything, just remove the default

nmahendru replied:
removed.

achristensen07 commented on LayoutTests/crypto/subtle/rsa-import-jwk-key-export-jwk-key-private.html:
```diff
@@ -54,6 +54,7 @@
     finishJSTest();
 });
 
+ 
```
This whitespace-only change doesn't seem like it improves anything.

nmahendru replied:
removed.

achristensen07 commented on Source/WebCore/crypto/gcrypt/CryptoKeyRSAGCrypt.cpp:
```diff
@@ -94,15 +94,9 @@ RefPtr<CryptoKeyRSA> CryptoKeyRSA::create(CryptoAlgorithmIdentifier identifier,
-    switch (keyData.type()) {
-    case CryptoKeyRSAComponents::Type::Public:
+    CryptoKeyType keyType = CryptoKeyType::Private;
+    if (keyData.isPublic())
```
Ditto.  This change doesn't seem meaningful.

nmahendru replied:
removed.

achristensen07 commented on Source/WebCore/crypto/keys/CryptoKeyRSAComponents.h:
```diff
@@ -72,6 +72,7 @@ class CryptoKeyRSAComponents {
     virtual ~CryptoKeyRSAComponents();
 
     Type type() const { return m_type; }
+    bool isPublic() const { return m_type == Type::Public; }
```
This function doesn't add anything that type() doesn't already have.

nmahendru replied:
removed.

nmahendru committed:
Enable some RSA key import export related tests.
rdar://115676486

Reviewed by Alex Christensen.

This updates the d value for RSA keys in various tests.
The way d was calculated changed around end of year 2019 which caused
the exported value to not match the input value.

* LayoutTests/crypto/workers/subtle/rsa-postMessage-worker.html:
* LayoutTests/crypto/subtle/rsa-import-jwk-key-export-jwk-key-private-expected.txt:
* LayoutTests/crypto/subtle/rsa-import-jwk-key-export-jwk-key-private.html:
* Source/WebCore/crypLayoutTests/platform/ios/TestExpectations
* LayoutTests/platform/mac/TestExpectations
* Source/WebCore/crypto/mac/CryptoKeyRSAMac.cpp
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBStructuredCloneBackwardCompatibility.mm
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBStructuredCloneBackwardCompatibilityRead.html
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBStructuredCloneBackwardCompatibilityWrite.html

Canonical link: https://commits.webkit.org/272691@main
